### PR TITLE
Add diamond database as datatype

### PR DESCRIPTION
**Pull request by bgruening**
ping @bebatut

### CODE REVIEW

**Comment by bebatut** [comment]
I tried to add a new class of binary (`galaxy.datatypes.binary:DMND`) and add a sniffer to detect this datatype. But I was not sure how to test it

**Comment by bgruening** [comment]
@galaxybot test this

**Comment by jmchilton** [comment]
Excellent - thanks @bebatut and @bgruening!

File: config/datatypes_conf.xml.sample [hunk 1]
```diff
@@ -38,6 +38,7 @@
     <datatype extension="len" type="galaxy.datatypes.chrominfo:ChromInfo" display_in_upload="true">
       <converter file="len_to_linecount.xml" target_datatype="linecount" />
     </datatype>
+    <datatype extension="dmnd" type="galaxy.datatypes.binary:Binary:DMND" subclass="True" display_in_upload="false"/>
```
Comment (mvdbeek) [inline config/datatypes_conf.xml.sample:41]: Is there a reason why this shouldn't be uploaded ? Given the sniffer is there we might just flip that.

**Pull request merged**